### PR TITLE
OOP school library: decorate a class

### PR DESCRIPTION
**Pull request by Lucas-Erkana**
**Create a class Nameable.**
Implement a method called correct_name that will raise a NotImplementedError.

**Turn Person class to Nameable**
Make sure that the Person class inherits from Nameable
Make sure that this class has a method correct_name implemented. It should simply return the name attribute.

**Prepare CapitalizeDecorator and TrimmerDecorator**
Create a class that inherits from the base Decorator class.
Implement a method correct_name that capitalizes the output of @nameable.correct_name.

**Create a class that inherits from the base Decorator class.**
Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

**Try the following code to check if it decorates the person**
person = Person.new(22, 'maximilianus')
  person.correct_name
  capitalizedPerson = CapitalizeDecorator.new(person)
  capitalizedPerson.correct_name
  capitalizedTrimmedPerson = TrimmerDecorator.new(capitalizedPerson)
  capitalizedTrimmedPerson.correct_name

image

### CODE REVIEW

File: capitalized_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class CapitalizedDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
```
Comment (OyePriscilla) [inline capitalized_decorator.rb]: - [ ] This class inherits from the `base decorator` class and not `nameable` in accordance to project requirement. 👍

File: README.md [hunk 2]
```diff
@@ -1,30 +1,39 @@
...
+# 📖 School Libray <a id="about-project"></a>
 
-**School Libray** is a library that manages all the students and teachers data.
+This project helps to understand the basics of Ruby programming language. It is a collection of exercises and projects that will help you to understand the basics of OOPS in Ruby programming language.
 
-### Tech Stack <a name="tech-stack"></a>
+### Tech Stack <a id="tech-stack"></a>
 
 <details>
 <summary>Server</summary>
```
Comment (OyePriscilla) [inline README.md]: - [ ] Please I observed that this link is redirecting me to your repo, kindly ensure that it redirects to appropriate page for a more engaging README file. 👍

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @Lucash2022 👍, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Create a classes and methods in Ruby ✔️ 
Implement inheritance in Ruby ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: capitalized_decorator.rb (hunk 1 above)
Comment (Lucas-Erkana) [reply capitalized_decorator.rb]: @OyePriscilla 
Thank you so much for pointing that out. I have resolved this [issue here ](https://github.com/Lucash2022/School-Library/blob/7f555b83667a072b52819ec9bca267c2b25dfcde/capitalized_decorator.rb#LL1C4-L1C4)

File: README.md (hunk 2 above)
Comment (Lucas-Erkana) [reply README.md]: @OyePriscilla 
Thank you. Problem resolved at this [line of code](https://github.com/Lucash2022/School-Library/blob/013c027d2385f211fa70deb7dc651c843008802b/README.md?plain=1#L70) plus at this [line as well](https://github.com/Lucash2022/School-Library/blob/013c027d2385f211fa70deb7dc651c843008802b/README.md?plain=1#L87)

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @Lucash2022,

✔️ Class decorator is well implemented

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**